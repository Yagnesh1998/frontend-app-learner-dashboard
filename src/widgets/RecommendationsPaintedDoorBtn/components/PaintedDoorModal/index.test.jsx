import { shallow } from '@edx/react-unit-test-utils';

import ModalView from '.';

jest.mock('../../track', () => ({
  trackPaintedDoorRecommendationHomeSkipBtnClicked: jest.fn(),
  trackPaintedDoorRecommendationHomeInterestBtnClicked: jest.fn(),
}));
jest.mock('@edx/paragon', () => ({
  ...jest.requireActual('@edx/paragon'),
  ModalDialog: {
    Body: 'ModalDialog.Body',
    Header: 'ModalDialog.Header',
    Title: 'ModalDialog.Title',
    Footer: 'ModalDialog.Footer',
    CloseButton: 'ModalDialog.CloseButton',
  },
  ActionRow: 'ActionRow',
}));

describe('ModalView', () => {
  const props = {
    isOpen: true,
    onClose: jest.fn(),
    variation: '',
  };
  describe('snapshot', () => {
    test('should renders default ModalView', () => {
      const wrapper = shallow(<ModalView {...props} />);
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
});
