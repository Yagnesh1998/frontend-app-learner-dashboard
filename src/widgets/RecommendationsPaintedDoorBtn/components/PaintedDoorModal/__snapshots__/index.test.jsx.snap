// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView snapshot should renders default ModalView 1`] = `
<div
  className="containers modal-container"
>
  <[object Object]
    hasCloseButton={false}
    isBlocking={true}
    isFullscreenScroll={true}
    isOpen={true}
    onClose={[MockFunction]}
    title=""
  >
    <ModalDialog.Header>
      <ModalDialog.Title
        className="mt-2"
      >
        Thank you for your interest!
      </ModalDialog.Title>
    </ModalDialog.Header>
    <ModalDialog.Body>
      <div>
        <p
          className="mt-2"
        >
          Personalized recommendations feature is not yet available. We are working hard on bringing it to your learner home in the near future.
        </p>
        <p>
          Would you like to be alerted when it becomes available?
        </p>
      </div>
    </ModalDialog.Body>
    <ModalDialog.Footer>
      <ActionRow>
        <ModalDialog.CloseButton
          onClick={[Function]}
          variant="tertiary"
        >
          Skip for now
        </ModalDialog.CloseButton>
        <ModalDialog.CloseButton
          onClick={[Function]}
          variant="primary"
        >
          Count me in!
        </ModalDialog.CloseButton>
      </ActionRow>
    </ModalDialog.Footer>
  </[object Object]>
</div>
`;
